### DISCUSSION TITLE
PullRequest mergeable_state possible values

### DISCUSSION THREAD

**Discussion by nickstanish** [community/community#24299] [Apps, API and Webhooks]
<p>What are the possible values for mergeable_state and associated descriptions?</p>
<p>There are some mentioned here: <a href="https://github.com/octokit/octokit.net/issues/1763" rel="nofollow noopener">https://github.com/octokit/octokit.net/issues/1763</a> but is that all of them? And are the descriptions accurate?</p>
<p>Mergeable is mentioned <a href="https://developer.github.com/v3/pulls/#response-1" rel="nofollow noopener">https://developer.github.com/v3/pulls/#response-1</a> but nothing about mergable_state</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>As it says in the issue you linked to, it’s an undocumented field. Sometimes our APIs will have undocumented fields but until they’re officially documented, they should be considered unofficial, in flux, may possibly go away, and could change at any time.</p>
<p>I’ll suggest to the engineering team that people are interested in documentation for the field being published, but I can’t promise a timeline or even if it will be officially documented.</p>
<p>Thanks for the question and feedback!</p>

**Reply by made2591** [reply]
is this field still available in early 2026, shall we stop using it?

**Comment by markcellus** [comment]
<aside class="quote group-github-employees" data-username="lee-dohm" data-post="2" data-topic="13926">
<div class="title">
<div class="quote-controls"></div>
 lee-dohm:</div>
<blockquote>
<p>Sometimes our APIs will have undocumented fields but until they’re officially documented, they should be considered unofficial, in flux, may possibly go away, and could change at any time.</p>
</blockquote>
</aside>
<p>Stumbled upon this page searching for the same request. But I’m not sure if this response is entirely helpful. Public fields exposed by APIs are assumed to always continue to fulfill their contracts regardless of whether or not those fields are documented. So I’d find it quite surprising that this is GitHub’s policy. Can you point me to where this policy is documented?</p>
<p>Another thing I’d like to point out is that while the <code>mergable_state</code> <em>values</em> aren’t documented, the field itself <em>is documented</em> because it shows up on the API documentation for the response to <a href="https://docs.github.com/en/free-pro-team@latest/rest/reference/pulls#get-a-pull-request" rel="noopener nofollow ugc">Get a pull request</a>. If the field is documented but the values aren’t, it’s misleading for it to even show up as an option in the API docs.</p>
<p>Thanks for any information you can provide.</p>

**Comment by WyriHaximus** [comment]
<p>Just FYI, the fields are documented on the GraphQL API: <a href="https://docs.github.com/en/graphql/reference/enums#mergestatestatus" class="inline-onebox" rel="noopener nofollow ugc">Enums - GitHub Docs</a></p>

**Reply by dewetblomerus** [reply]
This might be worse, several possible responses are documented in the GraphQL docs with no mention of the field being `undocumented` which might mislead someone into thinking all the possible responses are documented, which they are not.

I am seeing `unstable` as a response that does not show up in the GraphQL docs.

If it is undocumented, the documentation needs to reflect that it is `undocumented`. My request sounds like a contradiction, so maybe we need a new term for this, a `beta` field might more accurately describe fields like these where we can expect the documentation to be incomplete.
